activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -164,7 +164,7 @@ def resolve_all
         #   spec.config
         #   # => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" }
         #
-        def spec(config)
+        def spec(config, id = nil)
```

Comment:
I chose `id` because i wanted to be clear this is a unique identifier for the connection. When I think about `name` I doing thing that as been unique.